src/cq_cam/common.py

```diff
@@ -1,12 +1,270 @@
+M-code and G-code are further organised into modal groups.
+
+G-code Modal Groups
+- Group 0 - Non-modal codes: G4, G10 G28, G30, G52, G53, G92, G92.1, G92.2, G92.3 
+- Group 1 - Motion: G0, G1, G2, G3, G33, G38.n, G73, G76, G80, G81, G82, G83, G84, G85, G86, G87, G88, G89
+- Group 2 - Plane: G17, G18, G19, G17.1, G18.1, G19.1
+- Group 3 - Distance Mode: G90, G91
+- Group 4 - Arc Distance Mode: G90.1, G91.1
+- Group 5 - Feed Rate Mode: G93, G94, G95
+- Group 6 - Units: G20, G21
+- Group 7 - Cutter Diameter Compensation: G40, G41, G42, G41.1, G42.1
+- Group 8 - Tool Length Offset: G43, G43.1, G49
+- Group 10 - Canned Cycle Return Mode: G98, G99
+- Group 12 - Coordinate System: G54, G55, G56, G57, G58, G59, G59.1, G59.2, G59.3
+- Group 13 - Planner Control Mode: G61, G61.1, G64
+- Group 14 - Spindle Speed Mode: G96, G97
+
+M-code Modal Groups:
+- Group 4 - Stopping: M0, M1, M2, M30, M60
+- Group 7 - Spindle: M3, M4, M5
+- Group 8 - Coolant: M7, M8, M9
+
+In addition to the G and M address there are other letter addresses that are used in conjuction with them:
+- F - Feed Rate: G1, G2, G3
+- X - X Axis: G1, G2, G3
+- Y - Y Axis: G1, G2, G3
+- Z - Y Axis: G1, G2, G3
+- I - Arc Center in X Axis: G2, G3
+- J - Arc Center in Y Axis: G2, G3
+- K - Arc Center in Z Axis: G2, G3
+- R - Canned Cycle Arc Radius: G80, G81, G82, G83, G84, G85, G86, G88, G89
+- P - Canned Cycle Dwell Time: G80, G81, G82, G83, G84, G85, G86, G88, G89
+- L - Canned Cycle Loop Count: G80, G81, G82, G83, G84, G85, G86, G88, G89
+- S - Spindle: M3, M4
+- T - Tool Selection: M6
+- H - Tool Length Offset: G43
+- D - Radius Offset: G41, G42
+"""
+
+"""
+The G-codes have been organised differently from the proposed modal groups.
+
+## Motion Control
+- Paths: G00 (Rapid), G01 (Linear), G02 (Arc CW), G03 (Arc CCW), G04 (Pause/Dwell), G05 (Cubic Spline)
+- Home Position: G28 (Home 1), G30 (Home 2)
+- CannedCycle: G80 (Cancel), G81 (Simple), G82 (Simple Dwell), G83 (Peck), G84 (Tap), G98 (Retract to initial Z), G99 (Retruct to last Z)
+- CannedCycleReturnMode: G98, G99
+
+## Machine Configuration
+- Work Plane: G17 (XY), G18 (XZ), G19 (YZ)
+- Units: G20 (inch/min), G21 (mm/min)
+- Length Compensation: G43 (ON), G49 (OFF)
+- Radius Compensation: G40 (OFF), G41 (Left), G42 (Right)
+- WorkOffset: G53, G54, G55, G56, G57, G58, G59
+- MotionControlMode: G61 (Exact Stop Check), G64 (Blend)
+- DistanceMode: G90 (Absolute), G91 (Incremental)
+- ArcDistanceMode: G90.1 (Absolute), G91.1 (Incremental)
+- FeedRateControlMode: G93 (Inverse), G94 (Time), G95 (Revolution)
+- SpindleControlMode: G96 (Surface Speed), G97 (RPM)
+
+## Tool Configuration
+- LengthCompensation: G40 (OFF), G41 (Left), G42 (Right)
+- RadiusCompensation: G43 (ON), G49 (OFF)
+"""
+
+
+class Path(Enum):
+    RAPID = "G0"
+    LINEAR = "G1"
+    ARC_CW = "G2"
+    ARC_CCW = "G3"
+    PAUSE = "G4"
+    CUBIC = "G5"
+    QUADRATIC_SPLINE = "G5.1"
+    NURBS = "G5.2"
+
+    def to_gcode(self) -> str:
```

Comment:
While subclassing an enum with members is forbidden, it is OK to subclass an Enum that doesn't define any members ( https://docs.python.org/3/howto/enum.html#restricted-enum-subclassing )

Therefore since the to_gcode method is the same for all Enums in common.py, it should be possible to have a  base class such as

```python
class GCodeEnum(Enum):
    def to_gcode(self) -> str:
        return self._value_
```

and then just do

```python
class Path(GCodeEnum):
    ...
```